fundingmanager.go

```diff
@@ -1152,6 +1152,25 @@ func (f *fundingManager) handlePendingChannels(msg *pendingChansReq) {
 	msg.resp <- pendingChannels
 }
 
+// ProcessFundingMsg sends a message to the internal fundingManager goroutine,
+// allowing it to handle the lnwire.Message.
+func (f *fundingManager) ProcessFundingMsg(msg lnwire.Message, peer lnpeer.Peer) {
+	switch m := msg.(type) {
+	case *lnwire.OpenChannel:
+		f.processFundingOpen(m, peer)
+	case *lnwire.AcceptChannel:
+		f.processFundingAccept(m, peer)
+	case *lnwire.FundingCreated:
+		f.processFundingCreated(m, peer)
+	case *lnwire.FundingSigned:
+		f.processFundingSigned(m, peer)
+	case *lnwire.FundingLocked:
+		f.processFundingLocked(m, peer)
+	case *lnwire.Error:
+		f.processFundingError(m, peer.IdentityKey())
```

Comment:
should we error on unsupported message?

Reply:
I'm not sure I follow?

Reply:
I think the way I did this is not the most efficient, this duplicates the switch statement in the reservation coordinator and I think we can get away without it/

Reply:
yeah I think we should so as to not misuse the API

Reply:
I take it back - doing so would lead to a lot of code change in tests that I want to avoid in this PR. I mainly want to get rid of duplicated or confusing code.